server/routes/additional-learning-needs-screener/create/check-your-answers/checkYourAnswersController.ts

```diff
@@ -0,0 +1,63 @@
+import { NextFunction, Request, Response } from 'express'
+import type { ReferenceDataItemDto } from 'dto'
+import { asArray } from '../../../../utils/utils'
+
+export default class CheckYourAnswersController {
+  getCheckYourAnswersView = async (req: Request, res: Response, next: NextFunction) => {
+    const { challengesReferenceData, strengthsReferenceData } = res.locals
+    const { alnScreenerDto } = req.journeyData
+
+    const viewRenderArgs = {
+      screenerDate: alnScreenerDto.screenerDate,
+      challenges: mapSelectedTypesToCategories(alnScreenerDto.challenges, challengesReferenceData),
+      strengths: mapSelectedTypesToCategories(alnScreenerDto.strengths, strengthsReferenceData),
+      errorRecordingAlnScreener: req.flash('pageHasApiErrors')[0] != null,
+    }
+    return res.render('pages/additional-learning-needs-screener/check-your-answers/index', viewRenderArgs)
+  }
+}
+
+/**
```

Comment:
We need to do this mapping because the DTO (that contains the values the user has selected) only contains the actual selected types. It does not contain the category hierarchy, so we need this function to get the structure back again for the purpose of the nunjucks template.